Improve dev tools to update browser variables list.

shapkarin opened the pull request:
resolves #164 and also closes #166

shapkarin commented on globals.json:
```diff
@@ -227,20 +227,24 @@
 	"browser": {
 		"AbortController": false,
 		"AbortSignal": false,
-		"addEventListener": false,
```
not sure why it's removed

shapkarin commented:
That's what I trying to do:
- I cloned jshint and eslint and added it as `devDependencies`
- next I plan to add it to the `get-browser-globals.js` script
- in this way I can get the variables directly from the files
- not sure if they will be imported as well to the result module but we can try some build trick to make that works out

shapkarin commented on globals.json:
```diff
@@ -453,327 +347,69 @@
-		"ResizeObserverEntry": false,
 		"resizeTo": false,
 		"Response": false,
-		"RTCCertificate": false,
-		"RTCDataChannel": false,
-		"RTCDataChannelEvent": false,
-		"RTCDtlsTransport": false,
-		"RTCIceCandidate": false,
-		"RTCIceGatherer": false,
-		"RTCIceTransport": false,
-		"RTCPeerConnection": false,
-		"RTCPeerConnectionIceEvent": false,
-		"RTCRtpContributingSource": false,
-		"RTCRtpReceiver": false,
-		"RTCRtpSender": false,
-		"RTCSctpTransport": false,
-		"RTCSessionDescription": false,
-		"RTCStatsReport": false,
-		"RTCTrackEvent": false,
```
@sindresorhus  something is missing. I guess we need to combine that `jshint` import method with the previous one somehow.

shapkarin commented:
@sindresorhus 
I start to [automate](https://github.com/sindresorhus/globals/pull/165/commits/aad6d7a8e565d22229c68999ca15b86d0343120e) that somehow. At least for some steps from update's `readme.md`

### Usage
- Run `npm run get-jshint-browser`.
- Open an Incognito window in Chrome Canary and paste the above into the console.
- You'll now have a new object in your clipboard.
- Paste yours current clipboard to the field `my_browser` field at the root of the project.
- run `npm run combine-browser`
- Copy and paste the result from the `browser_vars.json` to the field `browser` at the `globals.json`

shapkarin commented on update/get-browser-globals/readme.md:
```diff
@@ -0,0 +1,7 @@
+### Usage
+- Run `npm run get-jshint-browser'.
```
```suggestion
- Run `npm run get-jshint-browser`.
```

shapkarin commented:
@sindresorhus I hope it's a good one contribution and that can help me to make my package better.
Maybe even the `eslint` as well. I will be wait the review and some response.

sindresorhus commented:
Can you describe how your PR is solving those referenced issues?

sindresorhus commented:
We definitely don't want to fetch globals directly from JSHint. That readme reference just points out where the global originally came from.

sindresorhus commented:
And there is already a similar PR open: https://github.com/sindresorhus/globals/pull/132 How is yours different?

shapkarin commented:
This PR does not contains global browser variables update tool set
refactoring. I added the tools that help to extract that list from jshint,
combine it with yours approach to get them from console log and reface it a
little to avoid the test problems.

вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com>:

> And there is already a similar PR open: #132
> <https://github.com/sindresorhus/globals/pull/132> How is yours different?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sindresorhus/globals/pull/165#issuecomment-657225162>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALFGLV3FEONSYC7SJAM3B3R3G6GVANCNFSM4ORCHVRQ>
> .
>

shapkarin commented:
Currently my toolset is WIP. Not ready but almost, only after the PR I
realised that. You can look at it also at that repo for more recent version
as I remember, at least for updated readme
https://github.com/shapkarin/global-diff/tree/combine/update/get-browser-globals

вс, 12 июля 2020 г. в 17:02, Юрий Шапкарин <yu.shapkarin@gmail.com>:

> This PR does not contains global browser variables update tool set
> refactoring. I added the tools that help to extract that list from jshint,
> combine it with yours approach to get them from console log and reface it a
> little to avoid the test problems.
>
> вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com>:
>
>> And there is already a similar PR open: #132
>> <https://github.com/sindresorhus/globals/pull/132> How is yours
>> different?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/sindresorhus/globals/pull/165#issuecomment-657225162>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AALFGLV3FEONSYC7SJAM3B3R3G6GVANCNFSM4ORCHVRQ>
>> .
>>
>

shapkarin commented:
> We definitely don't want to fetch globals directly from JSHint. That
readme reference just points out where the global originally came from.

JsHint is added as a devDependency so it’s not for the bundle. But for the
usage to the update the browsers var. As I remember I done that somehow
with it and updated list comes in the automated way. Not so wip as I
remember  I’m not with PC now, can check later.

Currently my toolset is WIP. Not ready but almost, only after the PR I
> realised that. You can look at it also at that repo for more recent version
> as I remember, at least for updated readme
>
> https://github.com/shapkarin/global-diff/tree/combine/update/get-browser-globals
>
> вс, 12 июля 2020 г. в 17:02, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>
>> This PR does not contains global browser variables update tool set
>> refactoring. I added the tools that help to extract that list from jshint,
>> combine it with yours approach to get them from console log and reface it a
>> little to avoid the test problems.
>>
>> вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com>:
>>
>>> And there is already a similar PR open: #132
>>> <https://github.com/sindresorhus/globals/pull/132> How is yours
>>> different?
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/sindresorhus/globals/pull/165#issuecomment-657225162>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AALFGLV3FEONSYC7SJAM3B3R3G6GVANCNFSM4ORCHVRQ>
>>> .
>>>
>>

shapkarin commented:
Sorry, messages are fragmented from the email answer. So I remember that -
I update the list with jshint + yours approach. And I save tools that I
used for this. And this or contains some version.

пн, 13 июля 2020 г. в 04:20, Юрий Шапкарин <yu.shapkarin@gmail.com>:

> > We definitely don't want to fetch globals directly from JSHint. That
> readme reference just points out where the global originally came from.
>
> JsHint is added as a devDependency so it’s not for the bundle. But for the
> usage to the update the browsers var. As I remember I done that somehow
> with it and updated list comes in the automated way. Not so wip as I
> remember  I’m not with PC now, can check later.
>
> Currently my toolset is WIP. Not ready but almost, only after the PR I
>> realised that. You can look at it also at that repo for more recent version
>> as I remember, at least for updated readme
>>
>> https://github.com/shapkarin/global-diff/tree/combine/update/get-browser-globals
>>
>> вс, 12 июля 2020 г. в 17:02, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>>
>>> This PR does not contains global browser variables update tool set
>>> refactoring. I added the tools that help to extract that list from jshint,
>>> combine it with yours approach to get them from console log and reface it a
>>> little to avoid the test problems.
>>>
>>> вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com>:
>>>
>>>> And there is already a similar PR open: #132
>>>> <https://github.com/sindresorhus/globals/pull/132> How is yours
>>>> different?
>>>>
>>>> —
>>>> You are receiving this because you authored the thread.
>>>> Reply to this email directly, view it on GitHub
>>>> <https://github.com/sindresorhus/globals/pull/165#issuecomment-657225162>,
>>>> or unsubscribe
>>>> <https://github.com/notifications/unsubscribe-auth/AALFGLV3FEONSYC7SJAM3B3R3G6GVANCNFSM4ORCHVRQ>
>>>> .
>>>>
>>>

shapkarin commented:
@sindresorhus At that PR
https://github.com/sindresorhus/globals/pull/132/files
I don’t see any see any automated script that get jshint vars that pass
your lint (test). and But, yes, I can change the devDepenecy from my repo
to some other.

пн, 13 июля 2020 г. в 04:25, Юрий Шапкарин <yu.shapkarin@gmail.com>:

> Sorry, messages are fragmented from the email answer. So I remember that -
> I update the list with jshint + yours approach. And I save tools that I
> used for this. And this or contains some version.
>
> пн, 13 июля 2020 г. в 04:20, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>
>> > We definitely don't want to fetch globals directly from JSHint. That
>> readme reference just points out where the global originally came from.
>>
>> JsHint is added as a devDependency so it’s not for the bundle. But for
>> the usage to the update the browsers var. As I remember I done that somehow
>> with it and updated list comes in the automated way. Not so wip as I
>> remember  I’m not with PC now, can check later.
>>
>> Currently my toolset is WIP. Not ready but almost, only after the PR I
>>> realised that. You can look at it also at that repo for more recent version
>>> as I remember, at least for updated readme
>>>
>>> https://github.com/shapkarin/global-diff/tree/combine/update/get-browser-globals
>>>
>>> вс, 12 июля 2020 г. в 17:02, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>>>
>>>> This PR does not contains global browser variables update tool set
>>>> refactoring. I added the tools that help to extract that list from jshint,
>>>> combine it with yours approach to get them from console log and reface it a
>>>> little to avoid the test problems.
>>>>
>>>> вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com>:
>>>>
>>>>> And there is already a similar PR open: #132
>>>>> <https://github.com/sindresorhus/globals/pull/132> How is yours
>>>>> different?
>>>>>
>>>>> —
>>>>> You are receiving this because you authored the thread.
>>>>> Reply to this email directly, view it on GitHub
>>>>> <https://github.com/sindresorhus/globals/pull/165#issuecomment-657225162>,
>>>>> or unsubscribe
>>>>> <https://github.com/notifications/unsubscribe-auth/AALFGLV3FEONSYC7SJAM3B3R3G6GVANCNFSM4ORCHVRQ>
>>>>> .
>>>>>
>>>>

shapkarin commented:
@sindresorhus It’s the way to get jshint vars and from Canary or other
manual, combine them and pass the tests.

пн, 13 июля 2020 г. в 04:36, Юрий Шапкарин <yu.shapkarin@gmail.com>:

> @sindresorhus At that PR
> https://github.com/sindresorhus/globals/pull/132/files
> I don’t see any see any automated script that get jshint vars that pass
> your lint (test). and But, yes, I can change the devDepenecy from my repo
> to some other.
>
> пн, 13 июля 2020 г. в 04:25, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>
>> Sorry, messages are fragmented from the email answer. So I remember that
>> - I update the list with jshint + yours approach. And I save tools that I
>> used for this. And this or contains some version.
>>
>> пн, 13 июля 2020 г. в 04:20, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>>
>>> > We definitely don't want to fetch globals directly from JSHint. That
>>> readme reference just points out where the global originally came from.
>>>
>>> JsHint is added as a devDependency so it’s not for the bundle. But for
>>> the usage to the update the browsers var. As I remember I done that somehow
>>> with it and updated list comes in the automated way. Not so wip as I
>>> remember  I’m not with PC now, can check later.
>>>
>>> Currently my toolset is WIP. Not ready but almost, only after the PR I
>>>> realised that. You can look at it also at that repo for more recent version
>>>> as I remember, at least for updated readme
>>>>
>>>> https://github.com/shapkarin/global-diff/tree/combine/update/get-browser-globals
>>>>
>>>> вс, 12 июля 2020 г. в 17:02, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>>>>
>>>>> This PR does not contains global browser variables update tool set
>>>>> refactoring. I added the tools that help to extract that list from jshint,
>>>>> combine it with yours approach to get them from console log and reface it a
>>>>> little to avoid the test problems.
>>>>>
>>>>> вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com>:
>>>>>
>>>>>> And there is already a similar PR open: #132
>>>>>> <https://github.com/sindresorhus/globals/pull/132> How is yours
>>>>>> different?
>>>>>>
>>>>>> —
>>>>>> You are receiving this because you authored the thread.
>>>>>> Reply to this email directly, view it on GitHub
>>>>>> <https://github.com/sindresorhus/globals/pull/165#issuecomment-657225162>,
>>>>>> or unsubscribe
>>>>>> <https://github.com/notifications/unsubscribe-auth/AALFGLV3FEONSYC7SJAM3B3R3G6GVANCNFSM4ORCHVRQ>
>>>>>> .
>>>>>>
>>>>>

shapkarin commented:
BTW thanks for your package and it’s basic update script, because it’s
inspire me to improve that. And that’s why I add that to you to your repo
As a PR and just an open an issue at the jshint’s one.


@sindresorhus It’s the way to get jshint vars and from Canary or other
> manual, combine them and pass the tests.
>
> пн, 13 июля 2020 г. в 04:36, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>
>> @sindresorhus At that PR
>> https://github.com/sindresorhus/globals/pull/132/files
>> I don’t see any see any automated script that get jshint vars that pass
>> your lint (test). and But, yes, I can change the devDepenecy from my repo
>> to some other.
>>
>> пн, 13 июля 2020 г. в 04:25, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>>
>>> Sorry, messages are fragmented from the email answer. So I remember that
>>> - I update the list with jshint + yours approach. And I save tools that I
>>> used for this. And this or contains some version.
>>>
>>> пн, 13 июля 2020 г. в 04:20, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>>>
>>>> > We definitely don't want to fetch globals directly from JSHint. That
>>>> readme reference just points out where the global originally came from.
>>>>
>>>> JsHint is added as a devDependency so it’s not for the bundle. But for
>>>> the usage to the update the browsers var. As I remember I done that somehow
>>>> with it and updated list comes in the automated way. Not so wip as I
>>>> remember  I’m not with PC now, can check later.
>>>>
>>>> Currently my toolset is WIP. Not ready but almost, only after the PR I
>>>>> realised that. You can look at it also at that repo for more recent version
>>>>> as I remember, at least for updated readme
>>>>>
>>>>> https://github.com/shapkarin/global-diff/tree/combine/update/get-browser-globals
>>>>>
>>>>> вс, 12 июля 2020 г. в 17:02, Юрий Шапкарин <yu.shapkarin@gmail.com>:
>>>>>
>>>>>> This PR does not contains global browser variables update tool set
>>>>>> refactoring. I added the tools that help to extract that list from jshint,
>>>>>> combine it with yours approach to get them from console log and reface it a
>>>>>> little to avoid the test problems.
>>>>>>
>>>>>> вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com
>>>>>> >:
>>>>>>
>>>>>>> And there is already a similar PR open: #132
>>>>>>> <https://github.com/sindresorhus/globals/pull/132> How is yours
>>>>>>> different?
>>>>>>>
>>>>>>> —
>>>>>>> You are receiving this because you authored the thread.
>>>>>>> Reply to this email directly, view it on GitHub
>>>>>>> <https://github.com/sindresorhus/globals/pull/165#issuecomment-657225162>,
>>>>>>> or unsubscribe
>>>>>>> <https://github.com/notifications/unsubscribe-auth/AALFGLV3FEONSYC7SJAM3B3R3G6GVANCNFSM4ORCHVRQ>
>>>>>>> .
>>>>>>>
>>>>>>

shapkarin commented:
@sindresorhus summary:
- jshint fork repo was imported as a devDependencies.
- this PR contains some Improve for dev tools to update the browser variables list. Not sure that it's not WIP.
- the same approach can be used for other environments.
- this PR contains browser variables list.

sindresorhus commented:
Again, we don't want to fetch from JSHint. The globals there are outdated. The whole point of this repo is to be the source of truth.

shapkarin commented:
OK, I got that. Maybe the reason that they want to keep all possible legacy support, not sure. I will close PR or later find some time to change the PR. Also it will be nice to change the `README.md` because it has some difference with yours response and create a misunderstanding.

sindresorhus commented:
I would be happy to make the readme clearer, but I don't know what part is unclear.

shapkarin commented:
> Extracted from JSHint and ESLint and merged.

shapkarin commented:
I can try to make it more clear maybe too. When I will back to my PR. Also plan to compare it with the other one.

shapkarin commented:
> Extracted from JSHint and ESLint and merged.

I decide that it's a normal process – extract it every time when you need to update + yours tool for browser vars. 

Also please check [that file](https://github.com/sindresorhus/globals/pull/165/files#diff-084045fe850b123ee962ff5af861219a) without that change I got an error at tests.

sindresorhus commented:
Closing for lack of activity and also because this is not the solution we wanted.
